### PR TITLE
feat(channels): adding hidden channels

### PR DESCRIPTION
**Pull request by joramirezStackit**

### CODE REVIEW

File: src/features/channels/components/channel-list/ChannelList.tsx [hunk 1]
```diff
@@ -1,31 +1,55 @@
...
+import { findChannels, getHiddenChannels, hideChannel } from '../../store/channel.actions';
 
 const ChannelList = () => {
+  const channelInitialState: Channel[] = [];
   const dispatch = useAppDispatch();
   const { channels, isLoading } = useAppSelector(selectState);
   const [selected, setSelected] = useState('');
+  const [currentChannels, setChannels] = useState(channelInitialState);
+
+  const getChannelList = () => {
```
Comment (mmiranda-stackit) [inline src/features/channels/components/channel-list/ChannelList.tsx:18]: We should leverage the business logic to the reducers. What I'd suggest to do, is to dispatch the getHiddenChannels action, create a hiddenChannels state variable in the channels slice, filter out the list, and use a selector to pull that state variable, this way, we would avoid too much login in the component and keep it clean, and we could potentially reuse that selector to other components as well.

File: src/features/channels/store/channel.actions.ts [hunk 2]
```diff
@@ -1,7 +1,36 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
+import { Channel } from '@metis/features/channels/types/channel';
 import channelService from '../services/channel.service';
 
 export const findChannels = createAsyncThunk('channels/findChannels', channelService.findChannels);
 export const createChannel = createAsyncThunk('channels/createChannel', channelService.create);
 
-export default { findChannels, createChannel };
+function hasStringJsonStructure(str: string | null) {
```
Comment (mmiranda-stackit) [inline src/features/channels/store/channel.actions.ts:8]: I suggest to take this utils function into a separate file.

File: src/features/channels/store/channel.actions.ts [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+    const result = JSON.parse(str);
+    const type = Object.prototype.toString.call(result);
+    return type === '[object Object]' || type === '[object Array]';
+  } catch (err) {
+    return false;
+  }
+}
+
+export const getHiddenChannels = () => {
+  const hiddenChannels = localStorage.getItem('@hiddenChannels');
```
Comment (mmiranda-stackit) [inline src/features/channels/store/channel.actions.ts:20]: I suggest to create a constant for '@hiddenChannels'

**Review by mmiranda-stackit — COMMENTED** [review]
Left some suggestions

**Review by josetheis-stackit — APPROVED** [review]
LGTM

**Pull request merged**